static/app/views/dashboardsV2/widgetBuilder/columnFields.tsx

```diff
@@ -39,17 +43,17 @@ export function ColumnFields({
     >
       {displayType === DisplayType.TABLE ? (
         <ColumnCollectionEdit
-          columns={columns}
+          columns={fields}
           onChange={onChange}
           fieldOptions={fieldOptions}
           organization={organization}
           source={widgetType}
         />
       ) : (
         <ColumnCollectionEdit
-          columns={columns.slice(0, columns.length - 1)}
+          columns={[...columns, ...aggregates.slice(0, aggregates.length - 1)]}
           onChange={newColumns => {
-            onChange([...newColumns, columns[columns.length - 1]]);
+            onChange([...newColumns, fields[fields.length - 1]]);
```

Comment:
Can I ask, what's the significance of `[fields.length - 1]` here?

This is when we're not in a table, and I think this component shows up in Top-N, so what I'm gathering is that the last index in fields is our Y-Axis for a Top-N widget and everything else shows up in the Columns section

Reply:
yep, you're right, it's how we get the y axis for top n